### ISSUE TITLE
Samsung TV: Video found but cannot be played (works on minidnla, trace attached)

### ISSUE THREAD

**Issue by probonopd** [anacrolix/dms#73]
Thanks for this useful piece of software.

UE46F... Series 6 Samsung TV shows icons for the videos served by dms in the DNLA browser, but trying to play the videos results in a black screen followed by an error message that the format is not supported "at this time". (So I think that this is different from https://github.com/anacrolix/dms/issues/56 where not even the icons can be seen.)

However, [minidnla](https://sourceforge.net/projects/minidlna/) can serve the **exact same video** files to the same Samsung TV just fine.

Attached are pcap traces of good (minidnla) and bad (dms) attempts to play the video on the Samsung TV.

[pcap.zip](https://github.com/anacrolix/dms/files/4404469/pcap.zip)

I noticed that minidnla sends quite some more DNLA-related HTTP headers than dms when serving the video, but there are other differences as well.

Please let me know how I can help further to test/debug this.

**Comment by anacrolix** [comment]
@probonopd thanks for the pcap! It looks like `dms` returns HTTP 500 errors for the SOAPACTION  "urn:schemas-upnp-org:service:ContentDirectory:1#X_GetFeatureList". I expect there could be a more appropriate error to return for this unsupported request, that may have the Samsung handle the results better. It may be aborting whatever it was attempting, rather than trying to proceed with an unsupported action. The other choice would be to implement the action.

I won't look into that just now, I'm not active on this project at the moment.

**Comment by probonopd** [comment]
Thanks @anacrolix for your analysis. Searching for `X_GetFeatureList` in Golang code, I found the [rclone](
https://github.com/rclone/rclone) project which, among many other features, can also serve directories on the local filesystem to DLNA devices. I can confirm that the Samsung TV __can__ play the same video using rclone indeed.

Looking at the rclone code, the solution may be rather easy:

https://github.com/rclone/rclone/blob/66e08e0cc8972000d84a61f262286f5644d9fdee/cmd/serve/dlna/cds.go#L325-L341

Where would this need to be plugged into the dms code?

**Comment by anacrolix** [comment]
@probonopd Thanks for finding this. I've submitted an issue to rclone, hopefully they are interesting in contributing their improvements on this. As for plugging it in, it would go alongside the code that was copied there in the first place 😆 .

**Comment by anacrolix** [comment]
@probonopd can you try #86?

**Comment by claes** [comment]
I have a Samsung UE46D6300 (so similar but not identical), and I can confirm this problem with v.1.6.0

**Comment by anacrolix** [comment]
> I have a Samsung UE46D6300 (so similar but not identical), and I can confirm this problem with v.1.6.0

@claes Did you try #86 ?

**Comment by claes** [comment]
@anacrolix No, but since it was merged I assumed it was part of 1.6.0, but perhaps it isn't?

**Comment by anacrolix** [comment]
You are quite right. I'll leave the issue open.

**Comment by WithoutPants** [comment]
Stash recently had a PR (https://github.com/stashapp/stash/pull/4784) submitted to fix what appears to be a related issue (https://github.com/stashapp/stash/issues/1768).

The change was to add the following headers when streaming a file:
- `transferMode.dlna.org` set to `Streaming`
- `contentFeatures.dlna.org` set to `DLNA.ORG_OP=01;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=01500000000000000000000000000000`

I can see that the former is already included upstream. Having had a look at https://github.com/anacrolix/dms/blob/6b89c448141baaa8a76ecf20ccb0780f33369b23/dlna/dlna.go#L35 it looks like the latter already corresponds to a `ContentFeatures` instance with `SupportTimeSeek = false`, `SupportRange = true`, and `Transcoded = false`. The only other difference is that the `DLNA.ORG_FLAGS` value starts with `015` instead of `017`. 

I don't have a Samsung TV to test with, nor do I know the DLNA protocol well enough to be confident that this change would be the solution to this original problem. A request was made to contribute the fix upstream, but as far as I can tell there doesn't appear to be anything useful to contribute.

**Comment by anacrolix** [comment]
@WithoutPants that's really helpful, appreciated.
